pkg/policy/repository.go

```diff
@@ -120,29 +120,53 @@ func (p *Repository) AllowsIngressLabelAccess(ctx *SearchContext) api.Decision {
 	return decision
 }
 
-// ResolveL4Policy resolves the L4 policy for a set of endpoints by searching
-// the policy repository for `PortRule` rules that are attached to a `Rule`
-// where the EndpointSelector matches `ctx.To`. `ctx.From` takes no effect and
+// ResolveL4IngressPolicy resolves the L4 ingress policy for a set of endpoints
+// by searching the policy repository for `PortRule` rules that are attached to
+// a `Rule` where the EndpointSelector matches `ctx.To`. `ctx.From` takes no effect and
 // is ignored in the search.  If multiple `PortRule` rules are found, all rules
 // are merged together. If rules contains overlapping port definitions, the first
 // rule found in the repository takes precedence.
 //
 // TODO: Coalesce l7 rules?
-func (p *Repository) ResolveL4Policy(ctx *SearchContext) (*L4Policy, error) {
+func (p *Repository) ResolveL4IngressPolicy(ctx *SearchContext) (*L4PolicyMap, error) {
 	result := NewL4Policy()
 
 	ctx.PolicyTrace("\n")
-	if ctx.EgressL4Only {
-		ctx.PolicyTrace("Resolving egress port policy for %+v\n", ctx.To)
-	} else if ctx.IngressL4Only {
-		ctx.PolicyTrace("Resolving ingress port policy for %+v\n", ctx.To)
-	} else {
-		ctx.PolicyTrace("Resolving port policy for %+v\n", ctx.To)
+	ctx.PolicyTrace("Resolving ingress port policy for %+v\n", ctx.To)
+
+	state := traceState{}
+	for _, r := range p.rules {
+		found, err := r.resolveL4IngressPolicy(ctx, &state, result)
+		if err != nil {
+			return nil, err
+		}
+		state.ruleID++
+		if found != nil {
+			state.matchedRules++
+		}
 	}
 
+	state.trace(p, ctx)
+	return &result.Ingress, nil
+}
+
+// ResolveL4EgressPolicy resolves the L4 egress policy for a set of endpoints
+// by searching the policy repository for `PortRule` rules that are attached to
+// a `Rule` where the EndpointSelector matches `ctx.From`. `ctx.To` takes no effect and
+// is ignored in the search.  If multiple `PortRule` rules are found, all rules
+// are merged together. If rules contains overlapping port definitions, the first
+// rule found in the repository takes precedence.
+//
+// TODO (ianvernon) unit tests
```

Comment:
Is this a blocking item for merging, or will it be followed up (+GH issue)?

Reply:
This was leftover, will remove; we now unit test coverage for this now.